src/controllers/auth-controller.ts

```diff
@@ -24,51 +24,64 @@ export default class AuthController extends AbstractController {
     this.router.post('/refresh', this.refresh);
 
     /* use auth.required to secure route */
-    this.router.get('/test', auth.required, this.test);
+    this.router.get('/test', authRequired, this.test);
 
     return this.router;
   }
 
-  private signUp = async (req: Request, res: Response, next: NextFunction): Promise<void> => {
+  private signUp = async (req: Request, res: Response, next: NextFunction) => {
     try {
+      const { email, password, firstName, lastName } = req.body;
+      if (!email || !password || !firstName || !lastName) {
+        return res.status(400).end();
+      }
+      const existUser = await this.users.findOne({ email });
```

Comment:
Probably it's redundant request, this checking is done by DB itself, you only have to catch such error returned from DB

Reply:
But how to check if that User exists?

Reply:
Just dont worry about. DB will fail if non uniq email will be passed. Be more lazy)

Reply:
fixed